auto_install/install.sh

```diff
@@ -1003,33 +1012,102 @@ confOVPN() {
+    $SUDO mkdir -p "/home/$pivpnUser/ovpns"
     $SUDO chmod 0777 -R "/home/$pivpnUser/ovpns"
 }
 
+finalExports() {
+    # Update variables in setupVars.conf file
+    if [ -e "${setupVars}" ]; then
+        sed -i.update.bak '/pivpnUser/d;/UNATTUPG/d;/pivpnInterface/d;/IPv4dns/d;/IPv4addr/d;/IPv4gw/d;/pivpnProto/d;/PORT/d;/ENCRYPT/d;/DOWNLOAD_DH_PARAM/d;/PUBLICDNS/d;/OVPNDNS1/d;/OVPNDNS2/d;/SERVER_NAME/d;' "${setupVars}"
+    fi
+    {
+        echo "pivpnUser=${pivpnUser}"
+        echo "UNATTUPG=${UNATTUPG}"
+        echo "pivpnInterface=${pivpnInterface}"
+        echo "IPv4dns=${IPv4dns}"
+        echo "IPv4addr=${IPv4addr}"
+        echo "IPv4gw=${IPv4gw}"
+        echo "pivpnProto=${pivpnProto}"
+        echo "PORT=${PORT}"
+        echo "ENCRYPT=${ENCRYPT}"
+        echo "DOWNLOAD_DH_PARAM=${DOWNLOAD_DH_PARAM}"
+        echo "PUBLICDNS=${PUBLICDNS}"
+        echo "OVPNDNS1=${OVPNDNS1}"
+        echo "OVPNDNS2=${OVPNDNS2}"
+        echo "SERVER_NAME=${SERVER_NAME}"
+    }>> "${setupVars}"
+}
+
+
+# I suggest replacing some of these names.
+
+#accountForRefactor() {
+#    # At some point in the future this list can be pruned, for now we'll need it to ensure updates don't break.
+#
+#    # Refactoring of install script has changed the name of a couple of variables. Sort them out here.
+#    sed -i 's/pivpnUser/PIVPN_USER/g' ${setupVars}
+#    #sed -i 's/UNATTUPG/UNATTUPG/g' ${setupVars}
+#    sed -i 's/pivpnInterface/PIVPN_INTERFACE/g' ${setupVars}
+#    sed -i 's/IPv4dns/IPV4_DNS/g' ${setupVars}
+#    sed -i 's/IPv4addr/IPV4_ADDRESS/g' ${setupVars}
+#    sed -i 's/IPv4gw/IPV4_GATEWAY/g' ${setupVars}
+#    sed -i 's/pivpnProto/TRANSPORT_LAYER/g' ${setupVars}
+#    #sed -i 's/PORT/PORT/g' ${setupVars}
+#    #sed -i 's/ENCRYPT/ENCRYPT/g' ${setupVars}
+#    #sed -i 's/DOWNLOAD_DH_PARAM/DOWNLOAD_DH_PARAM/g' ${setupVars}
+#    sed -i 's/PUBLICDNS/PUBLIC_DNS/g' ${setupVars}
+#    sed -i 's/OVPNDNS1/OVPN_DNS_1/g' ${setupVars}
+#    sed -i 's/OVPNDNS2/OVPN_DNS_2/g' ${setupVars}
+#    #sed -i 's/SERVER_NAME/SERVER_NAME/g' ${setupVars}
+#}
+
 installPiVPN() {
     stopServices
-    confUnattendedUpgrades
     $SUDO mkdir -p /etc/pivpn/
-    getGitFiles ${pivpnFilesDir} ${pivpnGitUrl}
+    confUnattendedUpgrades
     installScripts
     setCustomProto
     setCustomPort
     confOpenVPN
     confNetwork
+    SERVER_NAME="server"
     confOVPN
     setClientDNS
+    finalExports
+}
+
+updatePiVPN() {
+    #accountForRefactor
+    stopServices
+    confUnattendedUpgrades
+    installScripts
+    
+    # setCustomProto
+    echo "${pivpnProto}" > /tmp/pivpnPROTO
+    # write out the PROTO
+    PROTO=$pivpnProto
+    $SUDO cp /tmp/pivpnPROTO /etc/pivpn/INSTALL_PROTO
+    
+    #setCustomPort
+    # write out the port
+    echo ${PORT} > /tmp/INSTALL_PORT
+    $SUDO cp /tmp/INSTALL_PORT /etc/pivpn/INSTALL_PORT
+    
+    confOpenVPN
+    confNetwork
+    confOVPN
+    
+	# ?? Is this always OK? Also if you only select one DNS server ??
```

Comment:
Is this always OK? Also if you only select one DNS server, i.e. `OVPNDNS2` is empty?